### PR TITLE
sql: clean up wire protocol and protos

### PR DESCRIPTION
**Pull request by tamird**

### CODE REVIEW

File: sql/driver/wire.go [hunk 1]
```diff
@@ -26,43 +29,34 @@ const (
...
-	}
-	if d.IntVal != nil {
-		return strconv.FormatInt(*d.IntVal, 10)
-	}
-	if d.FloatVal != nil {
-		return strconv.FormatFloat(*d.FloatVal, 'g', -1, 64)
-	}
-	if d.BytesVal != nil {
-		return string(d.BytesVal)
+	val := d.GetValue()
```
Comment (petermattis) [inline sql/driver/wire.go]: Have you looked at the implementation of `GetValue()`? It just does the if-statements we were previously doing, but not you're doing those as well as a type switch. Perhaps something like the following:

```
  switch  {
    case d.BoolVal != nil:
      ...
    case d.IntVal != nil:
      ...
    case d.FloatVal != nil:
      ...
  }
```

**Comment by petermattis** [comment]
LGTM

File: sql/driver/wire.go (hunk 1 above)
Comment (tamird) [inline sql/driver/wire.go]: Yeah, GetValue() kinda sucks, but it's a closer interface to what we'll soon be using: https://github.com/golang/protobuf/commit/ef197e1a5f817199945aa5c661145b2c5f4b963f#diff-fe7ebfedd1e0c5013b2415edc0c10b17R157

File: sql/driver/wire.go [hunk 2]
```diff
@@ -26,43 +29,34 @@ const (
...
 
 // CreateReply creates an empty response for the request.
-func (*Request) CreateReply() *Response {
-	return &Response{}
-}
-
-// Header returns the response header.
-func (r *ResponseHeader) Header() *ResponseHeader {
-	return r
+func (Request) CreateReply() Response {
```
Comment (tschottdorf) [inline sql/driver/wire.go]: is there a reason to keep this?

Comment (tschottdorf) [inline sql/driver/wire.go]: oh, I see. There might be others than `Execute`.

File: sql/driver/wire.go (hunk 1 above)
Comment (petermattis) [inline sql/driver/wire.go]: Ok.

**Comment by vivekmenezes** [comment]
LGTM. Nice change

**Pull request merged**